### PR TITLE
Add faucet page - Closes #1072

### LINKED ISSUE #1072: Add reference page about the faucet plugin
Create a new page with information about the new faucet plugin in Lisk SDK 5.1.0

### PR DESCRIPTION
**Pull request by Tschakki**
Closes #1072

### CODE REVIEW

File: modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+== Plugin configuration
+
+The faucet plugin can be configured in the configuration of the blockchain application under `plugins.faucet`.
+
+[IMPORTANT]
+====
+There are three required config options that need to be defined for the faucet plugin to funciton correctly:
+
+. `encryptedPassphrase`: The passphrase of the genesis account which is encrypted with xref:{url_cryptography_encrypt_pw}[encryptPassphraseWithPassword()]
+. `captchaSecretkey`: The re-captcha secret key.
```
Comment (mitsuaki-u) [inline modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc]: Maybe we should add a link to the google recaptcha docs here to make it easier for users to find these keys.

File: modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+. `captchaSitekey`: The re-captcha site key.
+====
+
+.Dashboard options in the configuration
```
Comment (mitsuaki-u) [inline modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc]:
```suggestion
.Dashboard options in the configuration:
```

File: modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+== Plugin configuration
+
+The faucet plugin can be configured in the configuration of the blockchain application under `plugins.faucet`.
+
+[IMPORTANT]
+====
+There are three required config options that need to be defined for the faucet plugin to funciton correctly:
+
+. `encryptedPassphrase`: The passphrase of the genesis account which is encrypted with xref:{url_cryptography_encrypt_pw}[encryptPassphraseWithPassword()]
```
Comment (mitsuaki-u) [inline modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc:65]: this url is not working

File: modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+.Dashboard options in the configuration
+[source,js]
+----
+const { Application, FaucetPlugin, genesisBlockDevnet, configDevnet } = require('lisk-sdk');
+
+const appConfig = utils.objects.mergeDeep({}, configDevnet, {
+    plugins: {
+        faucet: {
+            encryptedPassphrase: "",
+            captchaSitekey: ""
```
Comment (mitsuaki-u) [inline modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc]:
```suggestion
            captchaSitekey: "",
```

File: modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc (hunk 2 above)
Comment (mitsuaki-u) [inline modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc]:
```suggestion
.Faucet options in the configuration
```

File: modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+  console.log("val:",val);
+});
+----
+
+==== Input
+
+----
+{
+    address: string,
+    token: string  (optional)
```
Comment (mitsuaki-u) [inline modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc]: token is required to get the funds

File: modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+
+|Actions
+a|
+* <<faucetauthorize>>
+* <<faucetfundtokens>>
+
+|Events
+|none
+
+|===
```
Comment (mitsuaki-u) [inline modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc:33]: The "Table of Contents" should be header/bolded

**Review by mitsuaki-u — CHANGES_REQUESTED** [review]

File: modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc (hunk 6 above)
Comment (Tschakki) [reply modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc:33]: this needs to be changed in the CSS of UI

File: modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc (hunk 3 above)
Comment (Tschakki) [reply modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc:65]: Will work on website, see: https://liskhq.github.io/lisk-docs/lisk-sdk/references/lisk-framework/faucet-plugin.html#plugin-configuration

File: modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+:toc: preamble
+:idprefix:
+:idseparator: -
+// URLs
+:url_recaptcha_test_keys: https://developers.google.com/recaptcha/docs/faq#id-like-to-run-automated-tests-with-recaptcha.-what-should-i-do
+:url_recaptcha: https://developers.google.com/recaptcha/intro
+// Project URLs
+:url_cryptography_encrypt_pw: references/lisk-elements/cryptography.adoc#encryptpassphrasewithpassword
+
+`@liskhq/lisk-framework-faucet-plugin` is a plugin for the Lisk Framework that provides a user interface to receive an certain amount of tokens on request.
```
Comment (chris529) [inline modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc:12]:
```suggestion
`@liskhq/lisk-framework-faucet-plugin` is a plugin for the Lisk Framework that provides a user interface that can receive a certain amount of tokens on request.
```

File: modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+The faucet plugin is a useful tool during development or in testing networks.
```
Comment (chris529) [inline modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc]:
```suggestion
The faucet plugin is a useful tool during development and also in testing networks.
```

File: modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc [hunk 9]
```diff
@@ -0,0 +1,250 @@
...
+
+The faucet plugin includes the following features:
+
+* enables to send tokens to different accounts from a faucet for testing
```
Comment (chris529) [inline modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc]:
```suggestion
* enables sending tokens to different accounts from a faucet for testing
```

File: modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc [hunk 10]
```diff
@@ -0,0 +1,250 @@
...
+* allows to customize the faucet UI with a custom logo and application URL
```
Comment (chris529) [inline modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc]:
```suggestion
* allows customizing the faucet UI with a custom logo and application URL
```

File: modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc [hunk 11]
```diff
@@ -0,0 +1,250 @@
...
+  });
+----
+
+== Plugin configuration
+
+The faucet plugin can be configured in the configuration of the blockchain application under `plugins.faucet`.
+
+[IMPORTANT]
+====
+There are three required config options that need to be defined for the faucet plugin to funciton correctly:
```
Comment (chris529) [inline modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc]:
```suggestion
There are three required config options that need to be defined in order for the faucet plugin to function correctly:
```

File: modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc [hunk 12]
```diff
@@ -0,0 +1,250 @@
...
+
+. `encryptedPassphrase`: The passphrase of the genesis account which is encrypted with xref:{url_cryptography_encrypt_pw}[encryptPassphraseWithPassword()]
+. `captchaSecretkey`: The re-captcha secret key. You can register here to get valid keys for recaptcha: {url_recaptcha}[^].
+ For testing purposes you don't need own keys, simply use the {url_recaptcha_test_keys}[test API keys recaptcha^].
```
Comment (chris529) [inline modules/ROOT/pages/references/lisk-framework/faucet-plugin.adoc]:
```suggestion
 For testing purposes, it is not necessary to use your own keys, simply use the {url_recaptcha_test_keys}[test API keys recaptcha^].
```

**Review by mitsuaki-u — DISMISSED** [review]

**Review by chris529 — COMMENTED** [review]
Looks good, just a few minor points 👍

**Pull request merged**